chapter3to8/src/main/java/webserver/Request.java

```diff
@@ -0,0 +1,117 @@
+package webserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import db.DataBase;
+import util.HttpRequestUtils;
+
+public class Request {
+	private String method;
+	private String url;
+	private String body;
+	private Map<String, String> cookies;
+	private boolean isLogined;
+	private Map<String, String> parameters;
+	private Map<String, String> header;
+	
+	public Request(BufferedReader in) throws IOException, NullPointerException{
+		String requestLine = in.readLine();
+		if(requestLine != null) {
+			String[] parts = requestLine.split(" ");
+			this.method = parts[0];
+			this.url = parts[1];
+			this.header = new HashMap<String, String>();
+			parseUrl(parts[1]);
+			readHeaders(in);
+		}
+		else {
```

Comment:
```suggestion
		throw new 다른 예외;
```
[제안] Request 생성자의 로직을 보면 else문이 없어도 똑같이 동작하고 있어요.
위 상황에서 if문이 끝나고 바로 예외가 터지도록 작성해보는게 어떨까요?
NullPointerException 또한 명시적으로 적어주지 않아도 requestLine이 null이라면 NullPointerException이 터지는 것 같아요. 

[질문] 위 제안에 이어서 널포인터 예외를 터뜨리지 않는다면 어떤 예외를 터뜨리는게 좋을까요?

[질문] 생성자에서 여러 로직들이 실행되고 있어요. 생성자에서 여러 로직을 실행하는게 맞을까요?

Reply:
메서드에 잘못된 인자를 전달하면 발생하는 예외로 IllegalArgumentException을 사용하는게 맞는 것 같네요. 그리고 생성자에서는 초기화의 역할을 하는 게 맞는 것 같은데 한 번 간단하게 바꿔보겠습니다.